### PR TITLE
Feature/integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by belalgamal46**
## Description
This pull request includes performance, functionality, and test coverage of the project:

- Added fix for n+1 problems to improve performance and reduce database queries
- Added capybara gem and configuration to enable integration testing of views
- Added view spec files to test the behavior of view components
- Added new tests to ensure proper functionality of views and improve test coverage
- Fixed a bug in the users_index test that was causing test failures
- Fixed a bug in fixing n+1 problem that was causing unexpected behavior
- Added test for user_show_spec.rb to improve test coverage
- Added default driver for Capybara to improve test stability
- Added test for the posts index page to ensure proper functionality of the posts index view
- Added test for posts show page to ensure proper functionality of the post show view
- Fixed linters to improve code quality and maintainability

These changes were made to improve the performance, functionality, and test coverage of the project, as well as to fix bugs and improve code quality.

## Additional Notes
- This pull request includes changes to the Gemfile and configuration files
- All tests are passing locally
- Linters have been fixed and the code follows best practices

### Before
before

### After
after

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,11 +1,12 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.where(author_id: params[:user_id]).paginate(page: params[:page], per_page: 2)
+    @posts = Post.includes(:author).where(author_id: params[:user_id]).paginate(page: params[:page], per_page: 2)
```
Comment (Owusu-Desmond) [inline app/controllers/posts_controller.rb]: - [ ] Please from the project requirement, you should make sure that the N+1 problem is solved when fetching all posts and their comments for a user so I think you should use `@posts = @user.posts.includes(:comments)` :pray:

File: spec/views/posts/posts_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  end
+
+  scenario 'Post content is visible' do
+    expect(page).to have_selector('.show-post-title h2', text: 'Test Post')
+    expect(page).to have_selector('.show-post-text p', text: 'Lorem ipsum dolor sit amet')
+    expect(page).to have_selector('.show-post-counters span', text: "Comments: #{post.comments_counter}")
+    expect(page).to have_selector('.show-post-counters span', text: "Likes: #{post.likes_counter}")
+    expect(page).to have_selector('.like-btn')
+
+    within('.show-post-comments') do
```
Comment (Owusu-Desmond) [inline spec/views/posts/posts_show_spec.rb]: - [ ] I suggest that you write Tests that asserts you to see who wrote the post as it is in the project requirement :blush:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @belalgamal46 and @Gmanlove , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
📌 Linters are passing ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Your README file is okay✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @belalgamal46, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**